src/main/java/uk/gov/justice/digital/hmpps/keyworker/Extensions.kt

```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.digital.hmpps.keyworker
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import kotlin.reflect.full.companionObject
+
+interface Logging
+
+inline fun <reified T : Logging> T.logger(): Logger = LoggerFactory.getLogger(getClassForLogging(T::class.java))
```

Comment:
Not sure this adds anything to the code base.  You still need a `companion object` anyway and just obfuscates what's going on